framework/src/timesteppers/IterationAdaptivePostprocessorDT.C

```diff
@@ -0,0 +1,526 @@
+  _sync_last_step(declareRestartableData<bool>("sync_last_step",false)),
+  _linear_iteration_ratio(isParamValid("linear_iteration_ratio") ? getParam<unsigned>("linear_iteration_ratio") : 25),  // Default to 25
+  _adaptive_timestepping(false),
+  _timestep_limiting_function(NULL),
+  _piecewise_timestep_limiting_function(NULL),
+  _times(0),
+  _max_function_change(-1),
+  _force_step_every_function_point(getParam<bool>("force_step_every_function_point")),
+  _tfunc_times(getParam<std::vector<Real> >("time_t").begin(), getParam<std::vector<Real> >("time_t").end()),
+  _time_ipol(getParam<std::vector<Real> >("time_t"),
+             getParam<std::vector<Real> >("time_dt")),
+  _use_time_ipol(_time_ipol.getSampleSize() > 0),
+  _growth_factor(getParam<Real>("growth_factor")),
+  _cutback_factor(getParam<Real>("cutback_factor")),
+  _nl_its(declareRestartableData<unsigned int>("nl_its", 0)),
+  _l_its(declareRestartableData<unsigned int>("l_its", 0)),
+  _cutback_occurred(declareRestartableData<bool>("cutback_occurred", false)),
+  _at_function_point(false),
+  _pps_value(getPostprocessorValue("postprocessor"))
+{
+
+  if (isParamValid("optimal_iterations"))
+  {
+    _adaptive_timestepping=true;
+    _optimal_iterations = getParam<int>("optimal_iterations");
+    if (isParamValid("iteration_window"))
+    {
+      _iteration_window = getParam<int>("iteration_window");
+    }
+    else
+    {
+      _iteration_window = ceil(_optimal_iterations/5.0);
+    }
+  }
+  else
+  {
+    if (isParamValid("iteration_window"))
+    {
+      mooseError("'optimal_iterations' must be used for 'iteration_window' to be used");
+    }
+    if (isParamValid("linear_iteration_ratio"))
+    {
+      mooseError("'optimal_iterations' must be used for 'linear_iteration_ratio' to be used");
+    }
+  }
+
+  if (isParamValid("timestep_limiting_function"))
+  {
+    _max_function_change = isParamValid("max_function_change") ?
+        getParam<Real>("max_function_change") : -1;
+  }
+  else
+  {
+    if (isParamValid("max_function_change"))
+    {
+      mooseError("'timestep_limiting_function' must be used for 'max_function_change' to be used");
+    }
+  }
+}
+
+IterationAdaptivePostprocessorDT::~IterationAdaptivePostprocessorDT()
+{
+}
+
+
+void
+IterationAdaptivePostprocessorDT::init()
+{
+  if (isParamValid("timestep_limiting_function"))
+  {
+    _timestep_limiting_function = &_fe_problem.getFunction(getParam<FunctionName>("timestep_limiting_function"), isParamValid("_tid") ? getParam<THREAD_ID>("_tid") : 0);
+    _piecewise_timestep_limiting_function = dynamic_cast<Piecewise*>(_timestep_limiting_function);
+    if (_piecewise_timestep_limiting_function)
+    {
+      unsigned int time_size = _piecewise_timestep_limiting_function->functionSize();
+      _times.resize(time_size);
+      for (unsigned int i=0; i < time_size; ++i)
+      {
+       _times[i] = _piecewise_timestep_limiting_function->domain(i);
+      }
```

Comment:
We discourage extraneous braces. They appear throughout your code. If you only have a single statement. Please drop the brackets